pkg/filter/extra_fields.go

```diff
@@ -0,0 +1,34 @@
+package filter
+
+import (
+	"github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ExtraFields carry on meta-information linking Tekton Run with Shipwright Build.
+type ExtraFields struct {
+	BuildRunNamespace string `json:"buildRunNamespace,omitempty"` // buildrun namespace
+	BuildRunName      string `json:"buildRunName,omitempty"`      // buildrun name
+}
+
+// IsEmpty checks if the BuildRunName is defined.
+func (e *ExtraFields) IsEmpty() bool {
+	return e.BuildRunName == ""
+}
+
+// GetNamespacedName returns a NamespacedName representation.
+func (e *ExtraFields) GetNamespacedName() types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: e.BuildRunNamespace,
+		Name:      e.BuildRunName,
+	}
+}
+
+// NewExtraFields instantiates a ExtraFields with informed BuildRun.
+func NewExtraFields(br *v1alpha1.BuildRun) ExtraFields {
+	return ExtraFields{
+		BuildRunNamespace: br.GetNamespace(),
```

Comment:
I think, the Run cannot reference an object from a different namespace. See https://tekton.dev/docs/pipelines/runs/#1-specifying-the-target-custom-task-with-ref. Based on that we will always create the BuildRun in the same namespace and can omit it here in the Run's status.

Reply:
You're correct, the golden rule is to always use the same namespace, however, for the sake of simplicity, is easier to get a `types.NamespacedName` from the `ExtraFields` given it contains namespace and name.

So in one hand it respects the "golden rule", and if that changes someday our controller continues to work as expected.